### PR TITLE
add production checklist to docs (close #2561)

### LINKED ISSUE #2561: document a guide or checklist for production deployment
Topics to be covered:
1) Using a reverse proxy with links to a nginx, caddy template
2) Command/Server flags to configure from [this](https://docs.hasura.io/1.0/graphql/manual/deployment/graphql-engine-flags/reference.html). 
   - Logging
   - Console availability
   - Admin endpoints to enable/disable
   - Allow-list
   (not exhaustive)
3) Optimisations (subscriptions) ?

@shahidhk Anything you'd want to add to the above?

### PR DESCRIPTION
**Pull request by marionschleifer**
### Description
A guide / checklist on what needs to be considered when moving to production.

### Affected components 

- Docs

### CODE REVIEW

File: docs/graphql/manual/guides/production-checklist.rst [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+^^^^^^^^^^
+
+You can `limit the number of rows <https://docs.hasura.io/1.0/graphql/manual/auth/authorization/permission-rules.html#row-fetch-limit>`_ that can be accessed with one request.
+
+Disable admin endpoints
+^^^^^^^^^^^^^^^^^^^^^^^
+
+It is recommended to `disable access to the schema / metadata API <https://docs.hasura.io/1.0/graphql/manual/api-reference/schema-metadata-api/index.html#disabling-schema-metadata-api>`_ for the production instance.
+
+Logging
```
Comment (dsandip) [inline docs/graphql/manual/guides/production-checklist.rst]: We should also have a note on disabling telemetry also. Something like "While PII data is not captured, it is recommended that telemetry be disabled for production instances." (with a link to the env var page)

File: docs/graphql/manual/guides/production-checklist.rst [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+--------
+
+The following steps are recommended to better secure your production GraphQL engine instance:
+
+Admin secret
+^^^^^^^^^^^^
+
+Configure an admin secret as described :doc:`here <../deployment/securing-graphql-endpoint>`.
+
+Reverse proxy & API gateway
```
Comment (coco98) [inline docs/graphql/manual/guides/production-checklist.rst]: We can change this section to just say Enable HTTPS. Let's not call it reverse proxy/API gateway. 
And the line can be: "Use a component or a service from your cloud vendor to enable HTTPS". Examples, can be nginx, caddy like they are.

File: docs/graphql/manual/guides/production-checklist.rst [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+`Here <https://caddyserver.com/docs/proxy>`_ is how you can configure a Caddy proxy.
+
+
+CORS domains
+^^^^^^^^^^^^
+
+Configure your CORS domains as described `here <https://docs.hasura.io/1.0/graphql/manual/deployment/graphql-engine-flags/config-examples.html#configure-cors>`_.
+
+API security
```
Comment (coco98) [inline docs/graphql/manual/guides/production-checklist.rst]: We should mention that people should review their authorization rules by going to the authorization role summary page. Especially they should double check permissions given to any anonymous type roles.

**Comment by rikinsk** [comment]
Also would like one change. As lot of people ask us about https, we should pull out the content of that section to a separate page in `deploying` and just link to it from the production checklist page like we have done with other sections

**Comment by rikinsk** [comment]
The section titles would be neater if they were all in the imperative form

- `Set admin secret`
- `Verify permissions`
- `Disable backend APIs` 
- `Disable Console`
- `Set up a query allow list`
- `Restrict CORS domains`
- `Enable HTTPS`
- `Configure logging`

**Comment by shahidhk** [comment]
@rikinsk made the changes you requested.

**Pull request merged**